libsolidity/AST.cpp

```diff
@@ -719,6 +790,51 @@ void Identifier::checkTypeRequirements()
 		BOOST_THROW_EXCEPTION(createTypeError("Declaration referenced before type could be determined."));
 }
 
+Declaration const* Identifier::overloadResolution(FunctionCall const& _functionCall)
+{
+	solAssert(m_overloadedDeclarations.size() > 1, "FunctionIdentifier not resolved.");
+	solAssert(!m_referencedDeclaration, "Referenced declaration should be null before overload resolution.");
+
+	std::vector<ASTPointer<Expression const>> arguments = _functionCall.getArguments();
+	std::vector<ASTPointer<ASTString>> const& argumentNames = _functionCall.getNames();
+
+	if (argumentNames.empty())
+	{
+		// positional arguments
+		std::vector<Declaration const*> possibles;
+		for (Declaration const* declaration: m_overloadedDeclarations)
+		{
+			TypePointer const& function = declaration->getType();
+			auto const& functionType = dynamic_cast<FunctionType const&>(*function);
+			TypePointers const& parameterTypes = functionType.getParameterTypes();
+
+			if (functionType.takesArbitraryParameters() ||
+				(arguments.size() == parameterTypes.size() &&
+				 std::equal(arguments.cbegin(), arguments.cend(), parameterTypes.cbegin(),
+							[](ASTPointer<Expression const> const& argument, TypePointer const& parameterType)
+							{
+								return argument->getType()->isImplicitlyConvertibleTo(*parameterType);
+							})))
+				possibles.push_back(declaration);			
+		}
+		if (possibles.empty())
+			BOOST_THROW_EXCEPTION(createTypeError("Can't resolve identifier"));
+		else if (std::none_of(possibles.cbegin() + 1, possibles.cend(),
```

Comment:
Could you add a comment to exlpain this rule? I am not sure if we really want it.